apps/mobile/src/screens/chat/components/categories-form.tsx

```diff
@@ -0,0 +1,118 @@
+import React from "react";
+
+import { Button, Checkbox, Text, View } from "~/libs/components/components";
+import {
+	useAppDispatch,
+	useAppForm,
+	useAppSelector,
+	useCallback,
+	useEffect,
+	useFormController,
+	useState,
+} from "~/libs/hooks/hooks";
+import {
+	DEFAULT_ERROR_MESSAGE,
+	toastMessage,
+} from "~/libs/packages/toast-message/toast-message";
+import { globalStyles } from "~/libs/styles/styles";
+import { quizCategoriesValidationSchema } from "~/packages/quiz/quiz";
+import { actions as quizActions } from "~/slices/quiz/quiz";
+
+const CategoriesForm: React.FC = () => {
+	const dispatch = useAppDispatch();
+	const { categories } = useAppSelector((state) => state.quiz);
+
+	const ZERO_CATEGORIES = 0;
+	const hasCategories = Boolean(categories.length > ZERO_CATEGORIES);
+
+	useEffect(() => {
+		if (!hasCategories) {
+			void dispatch(quizActions.getQuizCategories());
+		}
+	}, [dispatch, hasCategories]);
+
+	const [submittedCategories, setSubmittedCategories] = useState<null | string>(
+		null,
+	);
+
+	const { control, errors, handleSubmit, setValue } = useAppForm({
+		defaultValues: {
+			categories: [] as number[],
+		},
+		validationSchema: quizCategoriesValidationSchema,
+	});
+
+	const { field } = useFormController({
+		control,
+		name: "categories",
+	});
+
+	const selectedCategories = field.value;
+
+	const onSubmit = (): void => {
+		const selectedLabels = categories
+			.filter((cat) => selectedCategories.includes(cat.id))
+			.map((cat) => cat.name)
+			.join(", ");
+
+		setSubmittedCategories(selectedLabels || null);
+	};
+
+	const handlePress = (): void => {
+		void handleSubmit(onSubmit)();
+	};
+
+	const handleCheckboxChange = useCallback(
+		(categoryId: number): void => {
+			const updatedCategories = selectedCategories.includes(categoryId)
+				? selectedCategories.filter((id) => id !== categoryId)
+				: [...selectedCategories, categoryId];
+			setValue("categories", updatedCategories);
+		},
+		[selectedCategories, setValue],
+	);
+
```

Comment:
Suggestion not to use setValue directly. 
We can use onChange from useFormController everywhere.